src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -141,11 +145,41 @@ public function load(array $configs, ContainerBuilder $container)
 
     private function addTwigPath($twigFilesystemLoaderDefinition, $dir, $bundle)
     {
-        $name = $bundle;
+        $name = $this->normalizeBundleName($bundle);
+        $twigFilesystemLoaderDefinition->addMethodCall('addPath', array($dir, $name));
+    }
+
+    private function prependTwigPath($twigFilesystemLoaderDefinition, $dir, $bundle)
+    {
+        $name = $this->normalizeBundleName($bundle);
+        $twigFilesystemLoaderDefinition->addMethodCall('prependPath', array($dir, $name));
+    }
+
+    private function prependParentPaths($twigFilesystemLoaderDefinition, ContainerBuilder $container, $name, $parentName)
+    {
+        $bundleMetadata = $container->getParameter('kernel.bundles_metadata');
+        $parentBundle = $bundleMetadata[$parentName];
+
+        if (is_dir($dir = $container->getParameter('kernel.root_dir').'/Resources/'.$parentName.'/views')) {
+            $this->prependTwigPath($twigFilesystemLoaderDefinition, $dir, $name);
+        }
+
+        if (is_dir($dir = $parentBundle['path'].'/Resources/views')) {
```

Comment:
@xabbuh: `$bundleMetadata` contains the entire array of bundles, `$parentBundle` is correct I think?